tests/models_test.py

```diff
@@ -87,24 +94,39 @@ def test_run(self):
                     # ['aches', 'cough', 'fatigue', 'fever', 'gastro', 'loss_of_taste',
                     #  'mild', 'moderate', 'runny_nose', 'severe', 'trouble_breathing']
                     self.assertEqual(observed['reported_symptoms'].shape, (14, 12))
-                    if len(observed['candidate_encounters']):
+                    if observed['candidate_encounters'].size:
                         stats['humans'][h_i]['candidate_encounters_cnt'] += 1
-                        # candidate_encounters[:, 0] is the other human 4 bits id
-                        # candidate_encounters[:, 1] is the risk of getting contaminated during the encounter?
-                        # candidate_encounters[:, 2] is the number of days since the encounter
-                        self.assertEqual(observed['candidate_encounters'].shape[1], 3)
+                        # candidate_encounters[:, 0] is the other human 8 bits id
+                        # candidate_encounters[:, 1] is the new 4 bits risk of getting contaminated during the encounter
+                        # candidate_encounters[:, 2] is the 4 bits risk of getting contaminated during the encounter
+                        # candidate_encounters[:, 3] is the number of days since the encounter
+                        self.assertEqual(observed['candidate_encounters'].shape[1], 4)
                         self.assertGreaterEqual(observed['candidate_encounters'][:, 0].min(), 0)
-                        self.assertLess(observed['candidate_encounters'][:, 0].max(), 16)
-                        self.assertGreaterEqual(observed['candidate_encounters'][:, 1].min(), 0)
+                        self.assertLess(observed['candidate_encounters'][:, 0].max(), 256)
                         self.assertLess(observed['candidate_encounters'][:, 1].max(), 16)
+                        self.assertGreaterEqual(observed['candidate_encounters'][:, 1].min(), 0)
+                        # TODO: Tix this test which is failing because observed['candidate_encounters'][:, 2] as apparently
```

Comment:
Yea, that should never happen

Reply:
Last commit contains a quick attempt to fix this. I'm not sure to fully grasp the logic of the code yet tho so it might be wrong.